datalad/support/gitrepo.py

```diff
@@ -1545,7 +1545,7 @@ def commit(self, msg: Optional[str] = None,
 
         # handle interactive message entry by running another `git-commit`
         self._git_runner.run(
-            cmd + ['--amend', '--edit'],
+            ['git'] + cmd + ['--amend', '--edit'],
```

Comment:
yikes -- we nohow tested this since this is solely for interactive amending and we lazied out to make a test (e.g. may be could be done via overloading `VISUAL`  to something which would "serve" the editor etc), and all tested invocations returned above.

It seems that I am the one to blame for ruining it in c73033edf675a03655dba6ef3edac86b69371943 AKA `0.18.0~2^2~49^2~3` where in https://github.com/datalad/datalad/commit/c73033edf675a03655dba6ef3edac86b69371943#diff-bfc1c47d032fc72b15b6a79db3d49e4c4334cf96318c5efe48a3403744b32fb1L1376 I stripped away `self._git_cmd_prefix` from `cmd` since I started to use `self._call_git(` instead of `_git_runner.run(`  ... So indeed it would be correct to prefix, but likely we would want 

```suggestion
            self._git_cmd_prefix + cmd + ['--amend', '--edit'],
```
"just in case"